### PR TITLE
capz: add conformance jobs for master and release branches

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

Adding capz-based conformance jobs for master and release branches 

/assign @CecileRobertMichon

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh [hunk 1]
```diff
@@ -272,8 +274,8 @@ periodics:
       - --aksengine-template-url=https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/tests/k8s-azure/manifest/kubernetes.json
       - --aksengine-download-url=https://github.com/Azure/aks-engine/releases/download/nightly/aks-engine-nightly-linux-amd64.tar.gz
       # Specific test args
-      - --test_args=--ginkgo.focus=\[Conformance\] --ginkgo.skip=\[Serial\]
-      - --ginkgo-parallel=30
+      - --test_args=--ginkgo.focus=\[Conformance\]
+      - --ginkgo-parallel=1
```
Comment (chewong) [inline config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh:278]: Aligning ginkgo config with https://testgrid.k8s.io/sig-release-master-blocking#conformance-ga-only

File: config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh [hunk 2]
```diff
@@ -446,13 +448,51 @@ periodics:
...
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20210412-176e4b6-master
+      command:
+      - runner.sh
+      - ./scripts/ci-conformance.sh
+      env:
+      - name: E2E_ARGS
+        value: "-kubetest.use-ci-artifacts"
+      - name: KUBERNETES_VERSION
+        value: "${kubernetes_version}"
+      - name: CONFORMANCE_WORKER_MACHINE_COUNT
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh:476]: I might be wrong but I thought conformance required 5+ worker nodes?

Comment (chewong) [reply config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh:476]: I can't seem to find anything that mentions this requirement. The reason why I set it to 2 is that
- I want to minimize the core consumption
- existing conformance test on Azure are running on a 2-node cluster

Comment (CecileRobertMichon) [reply config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh:476]: SGTM, I might be misremembering

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**